proc_dash/app.py

```diff
@@ -324,66 +336,142 @@ def process_bagel(contents, filename):
+    dataset_summary = util.construct_summary_str(overview_df)
 
     return (
-        data.to_dict("records"),
+        overview_df.to_dict("records"),
+        pipelines_dict,
         None,
         session_opts,
+        {"display": "block"},
         "csv",
         dataset_summary,
         {"display": "block"},
     )
 
 
+@app.callback(
+    [
+        Output("pipeline-dropdown-container", "children"),
+        Output("interactive-datatable", "style_filter_conditional"),
+    ],
+    Input("memory-pipelines", "data"),
+    prevent_initial_call=True,
+)
+def create_pipeline_status_dropdowns(pipelines_dict):
+    """
+    Generates a dropdown filter with status options for each unique pipeline in the input csv,
+    and disables the native datatable filter UI for the corresponding columns in the datatable.
+    """
+    pipeline_dropdowns = []
+
+    if pipelines_dict is not None:
+        for pipeline in pipelines_dict:
+            new_pipeline_status_dropdown = dbc.Col(
+                [
+                    dbc.Label(
+                        pipeline,
+                        className="mb-0",
+                    ),
+                    dcc.Dropdown(
+                        id={
+                            "type": "pipeline-status-dropdown",
+                            "index": pipeline,
+                        },
+                        options=list(
+                            util.PIPE_COMPLETE_STATUS_SHORT_DESC.keys()
+                        ),
+                        placeholder="Select status to filter for",
+                    ),
+                ]
+            )
+            pipeline_dropdowns.append(new_pipeline_status_dropdown)
+
+        # "session" column filter is also disabled due to implemented dropdown filters for session
+        style_disabled_filters = [
+            {
+                "if": {"column_id": c},
+                "pointer-events": "None",
+            }
+            for c in list(pipelines_dict.keys()) + ["session"]
+        ]
+
+        return pipeline_dropdowns, style_disabled_filters
+
+    return pipeline_dropdowns, None
+
+
 @app.callback(
     [
         Output("interactive-datatable", "columns"),
         Output("interactive-datatable", "data"),
     ],
     [
-        Input("memory", "data"),
+        Input("memory-overview", "data"),
         Input("session-dropdown", "value"),
         Input("select-operator", "value"),
+        Input({"type": "pipeline-status-dropdown", "index": ALL}, "value"),
+        State("memory-pipelines", "data"),
     ],
 )
-def update_outputs(parsed_data, session_values, operator_value):
+def update_outputs(
+    parsed_data,
+    session_values,
+    session_operator,
+    status_values,
+    pipelines_dict,
+):
     if parsed_data is None:
         return None, None
 
     data = pd.DataFrame.from_dict(parsed_data)
 
-    if session_values:
-        data = util.filter_by_sessions(
+    if session_values or not all(v is None for v in status_values):
```

Comment:
:cherries: but I think 
```suggestion
    if session_values or any(v is not None for v in status_values):
```
is clearer